widget/list.go

```diff
@@ -724,3 +743,31 @@ func (l *listLayout) updateSeparators() {
 		l.separators[i].Show()
 	}
 }
+
+// invariant: visible is in ascending order of IDs
+func (l *listLayout) searchVisible(visible []itemAndID, id ListItemID) (*listItem, bool) {
+	// binary search
+	low := 0
+	high := len(visible) - 1
+	for low <= high {
+		mid := (low + high) / 2
+		if visible[mid].id == id {
+			return visible[mid].item, true
+		}
+		if visible[mid].id > id {
+			high = mid - 1
+		} else {
+			low = mid + 1
+		}
+	}
+	return nil, false
+}
```

Comment:
I wonder if it would be cleaner to just use `sort.Search()` here? Might be slightly slower but does it matter?

Reply:
Oh yeah, good call.

Reply:
Done